bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/RMUniversalRemoteHandler.java

```diff
@@ -0,0 +1,154 @@
+
+import static org.openhab.core.library.types.OnOffType.OFF;
+import static org.openhab.core.library.types.OnOffType.ON;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardOpenOption;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.broadlinkthermostat.internal.BroadlinkBindingConstants;
+import org.openhab.binding.broadlinkthermostat.internal.BroadlinkHandlerFactory;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.github.mob41.blapi.RM2Device;
+import com.github.mob41.blapi.mac.Mac;
+import com.github.mob41.blapi.pkt.cmd.rm2.SendDataCmdPayload;
+
+/**
+ * The {@link RMUniversalRemoteHandler} is responsible for handling RM Mini Universal Remotes.
+ *
+ * @author Miguel Álvarez - Initial contribution
+ */
+@NonNullByDefault
+public class RMUniversalRemoteHandler extends BroadlinkBaseHandler {
+    private final Logger logger = LoggerFactory.getLogger(RMUniversalRemoteHandler.class);
+    private @Nullable RM2Device rm2Device;
+
+    /**
+     * Creates a new instance of this class for the {@link RMUniversalRemoteHandler}.
+     *
+     * @param thing the thing that should be handled, not null
+     */
+    public RMUniversalRemoteHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * Initializes a new instance of a {@link RMUniversalRemoteHandler}.
+     */
+    @Override
+    public void initialize() {
+        super.initialize();
+        if (host != null && macAddress != null) {
+            try {
+                blDevice = new RM2Device(host, new Mac(macAddress));
+                this.rm2Device = (RM2Device) blDevice;
+                updateStatus(ThingStatus.ONLINE);
+            } catch (IOException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "Could not find broadlink universal remote device at host" + host + "with MAC+" + macAddress
+                                + ": " + e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Command: {}", command.toFullString());
+        authenticate(false);
+
+        if (command == RefreshType.REFRESH) {
+            logger.warn("Nothing to refresh on thing {}", thing.getUID());
+            return;
+        }
+        String channelId = channelUID.getIdWithoutGroup();
+        try {
+            switch (channelId) {
+                case BroadlinkBindingConstants.LEARNING_MODE:
+                    handleLearningCommand();
```

Comment:
It looks like only ON command is supported ?
In that case, you should call handleCommand only if command is ON.